src/bin/latency.rs

```diff
@@ -134,55 +138,42 @@ fn do_server(iface_name: String, oneway: bool) {
         },
         false => None,
     };
-
+    let mut rx_timestamp;
     while unsafe { RUNNING } {
-        let mut rx_timestamp;
-
         // TODO: Cleanup this code
         let recv_bytes = {
             match (oneway, msg) {
-                (true, Some(mut msg)) => match sock.recv_msg(&mut msg) {
-                    Ok(size) => {
-                        rx_timestamp = SystemTime::now();
-                        if size == 0 {
-                            eprintln!("????");
-                            continue;
-                        }
-                        size
-                    }
-                    Err(e) => {
-                        eprintln!("Failed to recv msg: {}", e);
+                (true, Some(mut msg)) => {
+                    let res = unsafe { libc::recvmsg(sock.fd, &mut msg, 0) };
+                    if res == -1 {
+                        continue;
+                    } else if res == 0 {
+                        eprintln!("????");
                         continue;
                     }
-                },
+                    res
+                }
                 _ => match sock.recv(&mut packet) {
-                    Ok(size) => {
-                        rx_timestamp = SystemTime::now();
-                        size
-                    }
+                    Ok(size) => size,
                     Err(_) => {
                         continue;
                     }
                 },
             }
         };
-
+        rx_timestamp = SystemTime::now();
```

Comment:
This part could be some problem because C version of this was following order to make time error as small as possible

- Recv packet
- Store timestamp
- Check whether packet is fine
- Do the rest

But, I think this is good to go for now (as RX timestamp is available and our main part is using RX/TX timestamp

Reply:
If delete that, cause "Use of possibly uninitialized variable" error. So add somewhere then I think that line is best.